src/rpp/rpp/operators/merge.hpp

```diff
@@ -73,7 +79,7 @@ struct merge_on_next
 
         state->count_of_on_completed_needed.fetch_add(1, std::memory_order::relaxed);
 
-        new_observable.subscribe(create_subscriber_with_state<ValueType>(sub.get_subscription().make_child(),
+        new_observable.subscribe(create_subscriber_with_state<ValueType>(state->childs_subscriptions.make_child(),
```

Comment:
q: You have made a child subscription when constructing the state. Do you really need to make child of child subscription here?